pkgs/development/libraries/science/math/cudnn/generic.nix

```diff
@@ -66,20 +75,18 @@ in stdenv.mkDerivation {
     runHook postInstall
   '';
 
-  # Set RUNPATH so that libcuda in /run/opengl-driver(-32)/lib can be found.
-  # See the explanation in addOpenGLRunpath.
+  # Otherwise autoPatchelf forgets $ORIGIN
   postFixup = ''
-    for lib in $out/lib/lib*.so; do
-      addOpenGLRunpath $lib
-    done
+    patchelf $out/lib/libcudnn.so --add-needed libcudnn_cnn_infer.so
```

Comment:
I'm just going to build and test pytorch with this.
We can remove this later when autoPatchelf UX allows that

Reply:
The test is smth like

```python
#! /usr/bin/env nix-shell
#! nix-shell -i python -p "(import ./. { config = { allowUnfree = true; cudaSupport = true; }; }).python3.withPackages (ps: with ps; [ pytorch ])"

import torch

a = torch.randn(1, 3, 128, 128, device="cuda")
torch.nn.functional.conv2d(a, torch.randn(1, 3, 2, 2, device="cuda"))
```

Reply:
So, this is not crashing anymore.

Reply:
To be clear: my assumption here was that it's always `libcudnn.so` importing other libraries in `$out/lib`, and not the other way around. Only `libcudnn.so` ends up with `$out/lib` in its runpath. Before the PR all of the libraries knew about `$out/lib`, but that may have been redundant.

The test verifies this assumption in a very simplistic setting, but I'm confident enough that this doesn't break things